source/guides/postgres-migration.rst

```diff
@@ -0,0 +1,199 @@
+
+Full-text indexes
+~~~~~~~~~~~~~~~~~
+
+It's possible that some words in the ``Posts`` and ``FileInfo`` tables can exceed the `limits of the maximum token length <https://www.postgresql.org/docs/11/textsearch-limitations.html>`__ for full text search indexing. In these cases, we recommend dropping the ``idx_posts_message_txt`` and ``idx_fileinfo_content_txt`` indexes from the PostgreSQL schema, and creating these indexes after the migration by running following queries:
+
+To drop indexes, run the following commands before the migration:
+
+.. code:: sql
+
+   DROP INDEX IF EXISTS idx_posts_message_txt;
+   DROP INDEX IF EXISTS idx_fileinfo_content_txt;
+
+To re-create indexes, run the following once the migration is completed:
+
+.. code:: sql
+
+   CREATE INDEX IF NOT EXISTS idx_posts_message_txt ON posts USING gin(to_tsvector('english', message));
+   CREATE INDEX IF NOT EXISTS idx_fileinfo_content_txt ON fileinfo USING gin(to_tsvector('english', content));
+
+Migrate the data
+----------------
+
+Once we set the schema to desired state, we can start migrating the **data** by running ``pgLoader`` \*\*
+
+\*\* Use the following configuration for the baseline of the data migration:
+
+.. code::
+
+   LOAD DATABASE
+        FROM      mysql://{{ .mysql_user }}:{{ .mysql_password }}@mysql:3306/{{ .source_schema }}
+        INTO      pgsql://{{ .pg_user }}:{{ .pg_password }}@postgres:5432/{{ .target_schema }}
+
+   WITH data only,
+        workers = 8, concurrency = 1,
+        multiple readers per thread, rows per range = 50000,
+        create no tables,
+        create no indexes,
+        preserve index names
+
+   SET PostgreSQL PARAMETERS
+        maintenance_work_mem to '128MB',
+        work_mem to '12MB'
+
+   SET MySQL PARAMETERS
+         net_read_timeout  = '120',
+         net_write_timeout = '120'
+
+   CAST column Channels.Type to channel_type drop typemod,
+        column Teams.Type to team_type drop typemod,
+        column UploadSessions.Type to upload_session_type drop typemod,
+        column Drafts.Priority to text,
+        type int when (= precision 11) to integer drop typemod,
+        type bigint when (= precision 20) to bigint drop typemod,
+        type text to varchar drop typemod,
+        type tinyint when (<= precision 4) to boolean using tinyint-to-boolean,
+        type json to jsonb drop typemod
+
+   MATERIALIZE VIEWS exclude_products
+        excluding table names matching ~<IR_>, ~<focalboard>
+
+   BEFORE LOAD DO
+        $$ ALTER SCHEMA public RENAME TO {{ .source_schema }}; $$
+
+   AFTER LOAD DO
+        $$ UPDATE {{ .source_schema }}.db_migrations set name='add_createat_to_teamembers' where version=92; $$,
+        $$ ALTER SCHEMA {{ .source_schema }} RENAME TO public; $$;
+
+Once you save this configuration file, eg. ``migration.load``, you can run the ``pgLoader`` with following command:
+
+.. code:: bash
+
+   pgLoader migration.load > migration.log
+
```

Comment:
Do we want to place a note here to remind users to run the commands to re-create the indexes as noted in lines 113/114?

Reply:
Sure, I can add that.